components/text_recognition/browser/text_recognition_win.cc

```diff
@@ -0,0 +1,230 @@
+      RunOnUIThread(std::move(callback_run_on_ui_thread));
+      return;
+    }
+  }
+
+  ComPtr<ISoftwareBitmapStatics> bitmap_factory;
+  hr = GetActivationFactory<
+      ISoftwareBitmapStatics,
+      RuntimeClass_Windows_Graphics_Imaging_SoftwareBitmap>(&bitmap_factory);
+  if (FAILED(hr)) {
+    VLOG(2) << "ISoftwareBitmapStatics factory failed: "
+            << logging::SystemErrorCodeToString(hr);
+    RunOnUIThread(std::move(callback_run_on_ui_thread));
+    return;
+  }
+
+  auto recognizer = std::make_unique<TextRecognizerWin>(
+      std::move(ocr_engine), std::move(bitmap_factory));
+  auto* recognizer_ptr = recognizer.get();
+  recognizer_ptr->Detect(
+      image, base::BindOnce(
+                 [](std::unique_ptr<TextRecognizerWin> recognizer,
+                    base::OnceCallback<void(const std::vector<std::string>&)>
+                        callback_run_on_ui_thread,
+                    std::vector<shape_detection::mojom::TextDetectionResultPtr>
+                        result) {
+                   std::vector<std::string> detected_string;
+                   for (const auto& re : result) {
+                     detected_string.push_back(re->raw_value);
+                   }
+                   RunOnUIThread(std::move(callback_run_on_ui_thread),
+                                 detected_string);
+                 },
+                 std::move(recognizer), std::move(callback_run_on_ui_thread)));
+}
+
+std::vector<std::string> GetAvailableRecognizerLanguages() {
+  // Loads functions dynamically at runtime to prevent library dependencies.
+  if (!base::win::ResolveCoreWinRTDelayload()) {
+    VLOG(2) << "Failed loading functions from combase.dll";
+    return {};
+  }
+
+  ComPtr<IOcrEngineStatics> engine_factory;
+  HRESULT hr = GetActivationFactory<IOcrEngineStatics,
+                                    RuntimeClass_Windows_Media_Ocr_OcrEngine>(
+      &engine_factory);
+  if (FAILED(hr)) {
+    VLOG(2) << "IOcrEngineStatics factory failed: "
+            << logging::SystemErrorCodeToString(hr);
+    return {};
+  }
+
+  ComPtr<IVectorView<Language*>> available_languages;
+  hr = engine_factory->get_AvailableRecognizerLanguages(&available_languages);
+  if (FAILED(hr)) {
+    VLOG(2) << "Fetch available languages failed: "
+            << logging::SystemErrorCodeToString(hr);
+    return {};
+  }
+
+  uint32_t lang_count;
+  hr = available_languages->get_Size(&lang_count);
+  if (FAILED(hr)) {
+    VLOG(2) << "get_Size failed: " << logging::SystemErrorCodeToString(hr);
+    return {};
+  }
+
+  WCHAR name[128]{};
+  ULONG nameLen = std::size(name);
+  ULONG langs = 0;
+  GetUserPreferredUILanguages(MUI_LANGUAGE_NAME, &langs,
+                              reinterpret_cast<PZZWSTR>(&name), &nameLen);
+  const std::string default_language =
+      brave_l10n::GetISOLanguageCode(base::WideToUTF8(std::wstring(name)));
+
+  // Locate default language at 0.
+  std::vector<std::string> language_codes = {default_language};
```

Comment:
can `default_language` be unsupported by OCR? Maybe we should add all supported languages and then move `default_language` to the 0 index if it exists in `available_languages`?